src/transformers/models/fsmt/modeling_fsmt.py

```diff
@@ -666,7 +666,9 @@ def forward(
         for idx, decoder_layer in enumerate(self.layers):
             # add LayerDrop (see https://arxiv.org/abs/1909.11556 for description)
             if output_hidden_states:
+                x = x.transpose(0, 1)
                 all_hidden_states += (x,)
+                x = x.transpose(0, 1)
```

Comment:
Is it possible to display the information about dimension changes here too? I think those comments are really helpful in the understanding of what's happening here.